redbot/core/settings_caches.py

```diff
@@ -238,19 +249,21 @@ def __init__(self, config: Config):
     async def remove_from_blacklist(self, guild: Optional[discord.Guild], role_or_user: List[int]):
         gid: Optional[int] = guild.id if guild else None
         role_or_user = role_or_user or []
-        if not isinstance(role_or_user, list) and not all(
-            isinstance(r_or_u, str) for r_or_u in role_or_user
+        if not isinstance(role_or_user, list) and all(
+            isinstance(r_or_u, int) for r_or_u in role_or_user
         ):
             raise TypeError("Blacklisted objects must be a list of ints")
+        if gid not in self._cached_blacklist:
+            self._cached_blacklist[gid] = []
         if gid is None:
             for obj_id in role_or_user:
-                if obj_id in self._cached_blacklist:
+                if obj_id in self._cached_blacklist[gid]:
                     self._cached_blacklist[gid].remove(obj_id)
                     async with self._config.blacklist() as curr_list:
                         curr_list.remove(obj_id)
         else:
             for obj_id in role_or_user:
-                if obj_id not in self._cached_blacklist:
+                if obj_id not in self._cached_blacklist[gid]:
```

Comment:
I missed this on first review:
```suggestion
                if obj_id in self._cached_blacklist[gid]:
```